util/file.go

```diff
@@ -321,7 +321,9 @@ func (manifest *fileManifest) clean(manifestPath string) error {
 				return errors.WithStackTrace(err)
 			}
 		} else {
-			if err := os.Remove(manifestEntry.Path); err != nil {
+			// use RemoveAll here instead of Remove because
+			// the first one doesn't return an error if the file doesn't exist
+			if err := os.RemoveAll(manifestEntry.Path); err != nil {
```

Comment:
But it also has the risk of deleting a folder and all its contents. Perhaps the right solution is to ignore "file doesn't exist" errors from `os.Remove`?

Reply:
I couldn't find another way to ignore this error. I expected `os.ErrNotExist` to work, but it doesn't https://play.golang.org/p/HB4u-EilFCu

There a risk of deleting a folder, but only if https://github.com/gruntwork-io/terragrunt/blob/e8030f6123c3490516202924d00b57c00ecc5692/util/file.go#L84 returns `false` for the directory.

Reply:
Yea, we're checking `IsDir` now, but after lots of refactoring, that may get lost. You can use `reflect.TypeOf` and good old `fmt.Sprintf` to print out details of the error so you can detect it correctly.

Reply:
fmt.Sprintf would make it unportable. Actually, I've checked how RemoveAll works and it uses https://golang.org/pkg/os/#IsNotExist which is exactly what we need!